viewer.py

```diff
@@ -417,8 +414,10 @@ def asset_loop(scheduler):
         else:
             logging.error('Unknown MimeType %s', mime)
 
-        if 'image' in mime or 'web' in mime:
+        if any(mimetype for mimetype in ['image', 'web', 'video']):
             duration = int(asset['duration'])
+            if 'video' in mime:
+                duration += 0.5 # Add a 500-ms delay to avoid cross-playback.
```

Comment:
Given the current changes in #1931, adding a small delay is just a workaround. There could be a way to communicate with the Qt binary to determine the current state of the player (e.g., stopped, playing, paused). The only problem at the moment is that getting the state from here gives `PlayingState` only for a very brief time. After a second, it gives `StoppedState`, even if the video's still playing.